spec/Functions.tex

```diff
@@ -221,8 +221,7 @@ \section{Procedure Definitions}
 The \sntx{return-type} is optional and is discussed in~\rsec{Return_Types}.
 A type function may not specify a return type.
 
-The optional \sntx{where-clause} is only applicable if the function is
-generic.  It is discussed in~\rsec{Where_Expressions}.
+The optional \sntx{where-clause} is discussed in~\rsec{Where_Expressions}.
```

Comment:
I don't want to push back, this was the feedback I was looking for with regard to the spec. I was pretty sure we wanted to move `22.4 -- "Where Expressions"` somewhere else, but I wasn't familiar enough with the spec to know where.